### PR TITLE
PedroDiez: Carrier_Billing_Checkout_API_documentation_and_yaml_update

### PR DESCRIPTION
**Pull request by PedroDiez**
## PR

Documentation for Carrier Billing Checkout API

Please see a proposal for Carrier Billing Checkout API documentation. Followed QoD API template as well. Source puml are provided for the illustration. Minor aligment of the yaml is also provided.

### CODE REVIEW

File: documentation/API_documentation/Carrier_Billing_Checkout_API.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+| --------------------------------------- |
+| **HTTP Request**<br> POST \<base-url>/carrier-billing-checkout/v0/purchases/{purchase_id}/payment/prepare<br>**Query Parameters**<br> No query parameters are defined.<br>**Path Parameters**<br> `purchase_id` MUST be present in the request, to identify the purchase process involved in the payment action. <br>**Request Body Parameters**<br> No request body parameters are defined.<br>
+ <br>**Response**<br> **200: Payment Triggered**<br> **Response body**: Information of triggered payment. <br> **id**: unique purchase identifier. Same value as in request path parameter.<br> **status**: purchase status.<br> **payment_otp**: provides the necessary information in order to validate an OTP, including the `validation_endpoint`. OTP code is sent offline to the user (e.g. via SMS), so end-user can input it in the endpoint provided. <br> **401:** **Unauthorized.** <br> **403:** **Permission Denied.** <br> **403:** **Accumulated Purchase Amount Overpassed.** <br> **404:** **Not Found.** <br> **409:** **Already Exists.** <br> **500:** **Server Error.**<br> **503:** **Service unavailable.** <br> **504:** **Request timeout exceeded.** |
```
Comment (bigludo7) [inline documentation/API_documentation/Carrier_Billing_Checkout_API.md:136]: To be sure to understand the mechanism. The server (carrier billing provider) sends a SMS to the enduser with a code (and a message). App request a code to the user. App send the code to the server (carrier billing provider)  to the validation endpoint.

If correct it probably worth it to shortly explain it.
This is probably not the only way to get end user approval right ? we can indicate this as one way to do it.

**Review by bigludo7 — COMMENTED** [review]
Thanks.
Few comments inline - no blocking point.

File: documentation/API_documentation/Carrier_Billing_Checkout_API.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+The Carrier Billing Checkout API enables an application to perform a purchase of a digital good (Content -e.g. Online Gaming related content-, Smart Mobility solutions -transport tickets, Event tickets, etc.-, Live Sports Streamings,...) and its associated payment guaranteeing security.
+
+The API also provides functionality to check the follow-up of a purchase and its lifecycle.
+
+Intented for a future phase:<br>
+* Details about a list of Purchases
+* Purchase subscriptions 
+* Notification Callbacks<br/>
+
```
[the remark this answers is not in the document]
Comment (PedroDiez) [reply documentation/API_documentation/Carrier_Billing_Checkout_API.md:16]: Yes, i am adding it

File: code/API_definitions/carrier_billing_checkout.yaml [hunk 3]
```diff
@@ -86,7 +86,7 @@ paths:
                 Generic403:
                   summary: Forbidden
                   value: 
-                    code: FORBIDDEN
+                    code: PERMISSION_DENIED
                     message: "Operation not allowed: ..."
```
[the remark this answers is not in the document]
Comment (PedroDiez) [reply code/API_definitions/carrier_billing_checkout.yaml:90]: My preference is to add in a later version, and i agree to have it in both proposals.

[the remark this answers is not in the document]
Comment (bigludo7) [reply code/API_definitions/carrier_billing_checkout.yaml:90]: OK ! works for me.

File: documentation/API_documentation/Carrier_Billing_Checkout_API.md (hunk 1 above)
Comment (PedroDiez) [reply documentation/API_documentation/Carrier_Billing_Checkout_API.md:136]: Yes, i am adding explanation. Not unique mechanism. Thanks for the comments!!

**Comment by PedroDiez** [comment]
@bigludo7 please take a look, have addressed all the comments provided

**Review by bigludo7 — APPROVED** [review]
Look good for me !

**Pull request merged**